core/src/main/scala/kafka/log/remote/RemoteLogManager.scala

```diff
@@ -592,18 +579,20 @@ class RemoteLogManager(fetchLog: TopicPartition => Option[Log],
       try {
         if (!isCancelled()) {
           //a. copy log segments to remote store
-          if (isLeader()) copyLogSegmentsToRemote()
-          else fetchLog(tp.topicPartition()).foreach { log =>
-            //todo-tier leader epoch to be computed here
-            val offset = remoteLogMetadataManager.highestOffsetForEpoch(tp, 0)
-            if (offset.isPresent) {
-              // todo-tier update remote index highest offset.
-              log.updateRemoteIndexHighestOffset(offset.get())
+          if (isLeader()) {
+            copyLogSegmentsToRemote()
+          } else {
+            val offset = findHighestRemoteOffset(tp)
+            fetchLog(tp.topicPartition()).foreach { log =>
+              if (offset.isPresent) {
+                // todo-tier update remote index highest offset.
+                log.updateRemoteIndexHighestOffset(offset.get())
+              }
             }
           }
 
-          //b. cleanup/delete expired remote segments
-          // handleExpiredRemoteLogSegments()
+          // b. cleanup/delete expired remote segments
+          handleExpiredRemoteLogSegments()
```

Comment:
Do both follower and leader attempt to remove expired remote log segments?  That doesn't seem right to me.